### DISCUSSION TITLE
I accidently setup tortoise to use the CPU - need help!

### DISCUSSION THREAD

**Discussion by user76765** [neonbjb/tortoise-tts#415] [Q&A]
Hello im an idiot - dont know what im doing,

rtx3060 laptop

I finally got tortoise-tts to work but looks like it is using the CPU. it took 5 minutes just for the simple phrase that is on this website ("I'm going to speak this")

I looked at task manager and my CPU usage was at 62% and GPU was 0-1/2%

so, looks like it is using the CPU.

How can i fix this? How can i reconfigr it to use the GPU? do i have to reinstall everything?

I have CUDA 11.7
i have installed pytorch as well (https://pytorch.org/get-started/locally/)
i also recently installed stablediffusion and that works fine (fast)

UPDATE:
uninstalled
>reinstalled inside a conda env
>installed pytorch
>got it to work again
>right now, trying to use a text file for the speech. and neither CPU or GPU is that high. CPU = 10% usage. GPU is hovering around 1-3%. memory is 39% lol

So, not sure whats going on. It seems faster than my first time. but im doing 2.37s/it. its take a while for a few sentences so far.

when i ran the single line - the intial sample were = 1.14s/it. but the processing afterwards were=10-11it/s

no idea if this is how its supposed to work

30 mins for a 790 character promt. with almost 0% GPU usage and 10% CPU

**Comment by ooofbrain** [comment]
yeah same problem. not as advanced gpu as you but mine is uding the cpu at 62% and the gpu is like 1-2. i got coda 11/7 too but right now its still using cpu. random.

**Reply by LucidFir** [reply]
I'd recommend everyone just delete their install and start again... way easier than spending days trying to figure it out.

**Reply by hejohns** [reply]
I just went through the process. It took some convincing to get pytorch to use cuda (finally just un/reinstalling, as suggested), but `python -m torch.utils.collect_env` that someone else mentioned was useful to figure it out.

**Comment by EricPanDev** [comment]
in my case I want the library to use the CPU. "cuda" is hardcoded in so many places in the library, which isnt very nice when you are trying to test things on a non cuda device, eg my M1 Macbook Air.

**Comment by neonbjb** [comment]
Happy to review PRs to fix this issue!

**Reply by EricPanDev** [reply]
I have this fork that works on m1, you could make a branch for it if you'd like https://github.com/EricPanDev/tortoise-tts-CPU

i would make PR but i dont really use that feature so not really sure how it works lol

**Reply by EricPanDev** [reply]
oop nvm just saw you already added it in change logs

**Comment by ShinoNia** [comment]
Hi. May I ask if you have been able to fix it? I also have the exact same problem

**Reply by LucidFir** [reply]
I'd recommend everyone just delete their install and start again... way easier than spending days trying to figure it out.

**Comment by LucidFir** [comment]
I'd recommend everyone just delete their install and start again... way easier than spending days trying to figure it out.
